### PR TITLE
feat: new advanced filter for course information

### LINKED ISSUE #1452: Major Information Attributes
Major information attributes would be helpful. This feature exists on Yale course search. For example, being able to filter polisci courses by comparative government, international relations, American government, etc. will be helpful when shopping for courses

https://coursetable.canny.io/admin/board/feature-requests/p/major-information-attributes

### PR DESCRIPTION
**Pull request by Etherite1**
should be fine, let's see

Fixes #1452

### CODE REVIEW

**Review by bearsyankees — COMMENTED** [review]
looks great, let's wait to see about mobile inclusion then can merge!

File: frontend/src/contexts/searchContext.tsx [hunk 1]
```diff
@@ -715,6 +734,17 @@ export function SearchProvider({ children }: { children: React.ReactNode }) {
         return false;
       }
 
+      if (
+        searchConfig.course_info_attributes.size !== 0 &&
+        listing.flag_info !== null &&
+        // !searchConfig.course_info_attributes.has(String(listing.flag_info))
+        Array.from(searchConfig.course_info_attributes).filter((value) =>
+          listing.flag_info.includes(value),
+        ).length == 0
```
Comment (Josh-Cena) [inline frontend/src/contexts/searchContext.tsx:743]:
```suggestion
        ).length === 0
```

File: frontend/src/utilities/constants.ts [hunk 2]
```diff
@@ -161,6 +161,776 @@ export const creditOptions = [
   { label: '2', value: 2 },
 ] as const;
 
+export const courseInfoAttributesOptions = [
```
Comment (Josh-Cena) [inline frontend/src/utilities/constants.ts:164]: Question for the future: is there a way to dynamically get this? Do we really need to statically code every single course attribute? They can easily get stale through time!

**Comment by Josh-Cena** [comment]
@Etherite1 If you have time, you may want to investigate emitting the info attributes data from API, sort of how `seasons.json` is handled at the moment.

File: frontend/src/utilities/constants.ts [hunk 3]
```diff
@@ -356,6 +356,341 @@ export const subjects: { [code: string]: string } = {
   ZULU: 'Zulu',
 };
 
+export const courseInfoAttributes = [
```
Comment (Josh-Cena) [inline frontend/src/utilities/constants.ts:359]:
```suggestion
// TODO: dynamically generate this like seasons.json
export const courseInfoAttributes = [
```

**Review by Josh-Cena — APPROVED** [review]
On demand we are going to merge this, but I'm going to investigate that later

**Pull request merged**